packages/core/admin/admin/src/utils/unique-admin-hash.js

```diff
@@ -0,0 +1,21 @@
+function bufferToHex(buffer) {
+  return [...new Uint8Array(buffer)].map((b) => b.toString(16).padStart(2, '0')).join('');
+}
+
+async function digestMessage(message) {
+  const msgUint8 = new TextEncoder().encode(message);
+  const hashBuffer = await crypto.subtle.digest('SHA-256', msgUint8);
+
+  return bufferToHex(hashBuffer);
+}
+
+export default async function hashAdminUserEmail(payload) {
```

Comment:
`generateAdminUserId` sounds like something identifiable. It needs to convey from the name that it's generating something that is not identifiable. Since we're an open source product, I don't want to add something that could theoretically upset users

Reply:
I don't think users that are upset about such a function wouldn't give the said function a look.
It's up to you but I basically think we have 2 choices:
- Either you transform this function to make it completely abstract, like a simple SHA 256 hash function that could be reused where needed - pretty straightforward but the name then needs to show that.
- Or, what I've talked about above (and actually what you did), the retrieval of the admin user ID logic is in its own function. My only thought is that the name shouldn't technically recap the inner working of a function. You could consider this nitpicking maybe 🤷